### PR TITLE
Feature/feature icon medal

### PR DESCRIPTION
**Pull request by oricat8**

### CODE REVIEW

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java [hunk 1]
```diff
@@ -50,17 +55,19 @@
 
     @BeforeEach
     public void setUp() {
-        medalEntity = new MedalEntity("Walker", new ArrayList<>(), "You have to walk a lot!");
+        medal = new Medal();
+        medal.setName("Walker");
+        medal.setDescription("You have to walk a lot!");
+        medal.setLevels(new ArrayList<>(Arrays.asList(5., 10., 25., 50., 100.)));
+        medal.setMedalIconPath(new byte[] {10, 55, 67 ,89, 103, 116});
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:62]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/MedalEntity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946&bid=18423000)

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java (hunk 1 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:62]: Codacy Issue found: [',' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946)

Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:62]: Codacy Issue found: ['10' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946)

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java [hunk 2]
```diff
@@ -50,17 +55,19 @@
 
     @BeforeEach
     public void setUp() {
-        medalEntity = new MedalEntity("Walker", new ArrayList<>(), "You have to walk a lot!");
+        medal = new Medal();
+        medal.setName("Walker");
+        medal.setDescription("You have to walk a lot!");
+        medal.setLevels(new ArrayList<>(Arrays.asList(5., 10., 25., 50., 100.)));
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:61]: Codacy Issue found: ['5.' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946)

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java (hunk 1 above)
Comment (marclll) [reply src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:62]: Fix this codacy issue

Comment (marclll) [reply src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java:62]: Fix this codacy issue

File: src/main/java/org/pesmypetcare/webservice/controller/medalmanager/MedalRestController.java [hunk 3]
```diff
@@ -23,6 +25,21 @@
     @Autowired
     private MedalService medalService;
 
+
```
Comment (marclll) [inline src/main/java/org/pesmypetcare/webservice/controller/medalmanager/MedalRestController.java:28]: Com es que  estas implementant la creació de medalles, no anava hardcoded?

**Review by marclll — CHANGES_REQUESTED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/service/medalmanager/MedalServiceImpl.java  -34
- src/main/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoImpl.java  -17
- src/main/java/org/pesmypetcare/webservice/controller/medalmanager/MedalRestController.java  -29
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/MedalEntity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946&bid=18423000)

**Review by marclll — DISMISSED** [review]

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java (hunk 1 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java]: Fix the comma placed wrong between 67 and 89.

**Review by santidrj — CHANGES_REQUESTED** [review]
There is a small code style issue

File: src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java (hunk 1 above)
Comment (oricat8) [reply src/test/java/org/pesmypetcare/webservice/dao/medalmanager/MedalDaoTest.java]: Done

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/medalmanager/MedalEntity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5623946&bid=18423000)

**Review by marclll — APPROVED** [review]

**Review by santidrj — APPROVED** [review]
:+1:

**Pull request merged**